server/etcdserver/api/etcdhttp/health.go

```diff
@@ -91,6 +98,31 @@ func NewHealthHandler(lg *zap.Logger, hfunc func(excludedAlarms AlarmSet, Serial
 	}
 }
 
+// newHealthHandler generates a http HandlerFunc for a health check function hfunc.
+func newHealthHandler(path string, lg *zap.Logger, hfunc func(*http.Request) (Health, error)) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			w.Header().Set("Allow", http.MethodGet)
+			http.Error(w, "Method Not Allowed", http.StatusMethodNotAllowed)
+			lg.Warn("/health error", zap.Int("status-code", http.StatusMethodNotAllowed))
+			return
+		}
+		h, err := hfunc(r)
+		if err != nil {
+			panic(err)
+		}
+		d, _ := json.Marshal(h)
```

Comment:
Sorry for not noticing it earlier. The proposed API is a strange mix of etcd and Kubernetes one. A mix of JSON and custom text format. 

Just see the response. I will be 
```
{
 "health": "true",
 "reason": "[-]data_corruption failed: Alarm active: CORRUPT\n[+]serializable_read ok\n",
}
```

This mix is worse than using a single format, as it requires to use both json and custom parser for client to get a failed check name. We should stick to either JSON or to Kubernetes text format. 

How Kubernetes response should look like:
```
[-]data_corruption failed: Alarm active: CORRUPT
[+]serializable_read
```

How pure json response should look like:
```
{
 "health": "true",
 "checks": {"
   data_corruption": {
     "status": "failed",
      "error": "Alarm active: CORRUPT"
    }, 
    "serializable_read": {
      "status": "ok",
    }
}
```

My recommendation is to use Kubernetes. Reason is that Etcd became part of Kubernetes and having a consistent format should make it easier to integrate for users vs custom format. If you like to support JSON, we could use [content type negotiation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation) as a way to serve Json to client that will request it.